syntax = "proto3";
package cafeteria;

enum Weekday {
        UNKNOWN_WEEKDAY = 0;
        MON = 1;
        TUE = 2;
        WED = 3;
        THU = 4;
        FRI = 5;
    }

    enum Cuisine {
        UNKNOWN_CUISINE = 0;
        POLISH = 1;
        ITALIAN = 2;
        INDIAN = 3;
        MEXICAN = 4;
    }

message Menu {
    int32 id= 1;
    Cuisine cuisine = 2;
    Weekday day =3;
    string dish_name = 4;
    string description = 5;
    repeated string ingredients = 6;
}

message CuisineSubscriptionRequest {
    Cuisine cuisine = 2;
}

message WeekdaySubscriptionRequest {
    Weekday weekday = 2;
}

service CafeteriaService{
    rpc SubscribeWeekday(WeekdaySubscriptionRequest) returns (stream Menu) {}

    rpc SubscribeCuisine(CuisineSubscriptionRequest) returns (stream Menu) {}
}